src/clients/SpokePoolClient.ts

```diff
@@ -656,6 +656,21 @@ export class SpokePoolClient extends BaseAbstractClient {
         }
 
         if (this.depositHashes[getRelayEventKey(deposit)] !== undefined) {
+          // Sanity check that this event is not a duplicate, even though the relay data hash is a duplicate.
+          const allDeposits = this._getDuplicateDeposits(deposit).concat(this.depositHashes[getRelayEventKey(deposit)]);
```

Comment:
I believe only need to check deposit hashes here since this is the block which assigns values to duplicate deposits.

Reply:
I'm not sure what you mean here?

Reply:
no its not. We only store a single deposit in `this.depositHashes` and any others get added to `this.duplicateDeposits`

Reply:
Ok yeah, I see. My mistake.